server/BraveAlerterConfigurator.js

```diff
@@ -99,6 +100,20 @@ class BraveAlerterConfigurator {
 
         // If the SMS came from the session's respondedByPhoneNumber
         if (alertSession.respondedByPhoneNumber === session.respondedByPhoneNumber) {
+          // If the client sends a request to reset, reject it by not updating the session as stored in the database
+          // NOTE: this *should* never happen, as the getClientMessageForRequestToReset function returns null
+          // Thus, the alert state machine should never transition from `STARTED` or `WAITING_FOR_REPLY` to `RESET`
+          if (alertSession.alertState === CHATBOT_STATE.RESET) {
```

Comment:
What do you think about doing this check before even starting the transaction?

And since this shouldn't ever happen, what do you think about adding a `logError` message if it does happen?

Reply:
Or, if you did do it after the beginning of the transaction, you could throw an error and then it will be caught and handled like other errors.

Reply:
The main reason I'm implementing it in this way is to have the replacement messages set to `null`... Though if an Error is thrown it wouldn't have much of a difference. I'll see about additing it before the transaction, and i'll use `logError`!